phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -1415,6 +1415,9 @@ private List<PeekingResultIterator> getIterators(List<List<Scan>> scan, Connecti
                             // Resubmit just this portion of work again
                             Scan oldScan = scanPair.getFirst();
                             byte[] startKey = oldScan.getAttribute(SCAN_ACTUAL_START_ROW);
+                            if (e2 instanceof StaleRegionBoundaryCacheException) {
```

Comment:
Sorry for the late comment. While I created the jira, I wanted to log all the exceptions coming from the server side not just StaleRegionBoundaryCacheException. I was thinking to logging the exception at L#1408, something like this.
```
SQLException sqle = ServerUtil.parseServerException(e);
LOG.warn("Exception encountered", sqle);
throw sqle;
```
Cc @Divneet18 @virajjasani @lokiore

Reply:
Not bad idea but i wonder whether this would result in redundant whole stacktrace logging. Perhaps, we can just log `e.getMessage()` and let this change to report whole stacktrace for `StaleRegionBoundaryCacheException` be as is?

Reply:
I don't think just printing `e.getMessage()`is a good idea. We should remove the current logging and add at L#1408. It would be duplicate for HashJoinCacheNotFoundException but thats ok.

Reply:
Removing from here is fine but adding full stacktraces might be overwhelming amount of logs is what i think. For every single SELECT query with some/any exception, we would log and that might be too much for logs.

It's fine though, let's give it a shot but I fear that during some genuine problem at the server side, we might be overwhelmed with long stacktraces that are on the other hand obvious errors and likely be ignored by anyone debugging.